### PR TITLE
Fix version class for greater than comparison

### LINKED ISSUE #2886: Version("6") < Version("6.0") == True
Version model has a bug.
Version("6") is considered lower than "6.0". It introduced a bug (fixed in #2885) with the default`std`.

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: closes #2886 
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.
- [x] I've opened another PR in the Conan docs repo to the ``develop`` branch, documenting this one. Also adding a description of the changes in the ``changelog.rst`` file. https://github.com/conan-io/docs

Missing changelog

### CODE REVIEW

File: conans/test/model/version_test.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ def simple_test(self):
         self.assertTrue(v1 < "1.11")
         self.assertTrue(v1 > "1.2")
         self.assertTrue(v1 > "1.2.2.2")
-        self.assertTrue(v1 < "1.2.3.2")
+        self.assertTrue(v1 > "1.2.3.2")
```
Comment (memsharded) [inline conans/test/model/version_test.py]: This is wrong 
Correct: ``1.2.3 < 1.2.3.2``

File: conans/model/version.py [hunk 2]
```diff
@@ -98,6 +98,26 @@ def __cmp__(self, other):
         if not isinstance(other, Version):
             other = Version(other)
 
+        # Check equals
```
Comment (lasote) [inline conans/model/version.py]: Probably the new code could be nicer like:

```
        # Check equals
        def get_el(a_list, index):
            if len(a_list) - 1 < index:
                return 0  # out of range, 4 == 4.0 == 4.0.0 
            return a_list[index]

        equals = all(get_el(other.as_list, ind) == get_el(self.as_list, ind)
                     for ind in range(0, max(len(other.as_list), len(self.as_list))))
        if equals:
            return 0
```

File: conans/model/version.py [hunk 3]
```diff
@@ -131,3 +149,12 @@ def __le__(self, other):
...
     def __ge__(self, other):
         return self.__cmp__(other) in [0, 1]
+
+    def __eq__(self, other):
+        return self.__cmp__(other) == 0
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __hash__(self):
```
[the remark this answers is not in the document]
Comment (lasote) [reply conans/model/version.py]: I think it is ok

File: conans/test/model/version_test.py [hunk 4]
```diff
@@ -39,6 +39,18 @@ def simple_test(self):
         self.assertTrue(Version("1.2.1-dev") > Version("1.2.alpha"))
         self.assertTrue(Version("1.2.1-dev") > Version("1.2-alpha"))
 
+        self.assertFalse(Version("4") < Version("4.0.0"))
+        self.assertFalse(Version("4") > Version("4.0.0"))
+        self.assertFalse(Version("4") != Version("4.0.0"))
+        self.assertTrue(Version("4") == Version("4.0.0"))
```
Comment (memsharded) [inline conans/test/model/version_test.py]: add test 4.0 == 4.0.0

**Pull request merged**